metrics/query.global/global.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2022 Block, Inc.
+
+package queryglobal
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/cashapp/blip"
+	"github.com/cashapp/blip/metrics/util"
+)
+
+const (
+	DOMAIN = "query.global"
+
+	OPT_RESPONSE_TIME_PERCENTILES = "response-time-percentiles"
+	OPT_RESPONSE_TIME_OPTIONAL    = "response-time-optional"
+	OPT_TRUNCATE_TABLE            = "truncate-table"
+	DEFAULT_PERCENTILE_OPTION     = "999"
+
+	BASE_QUERY     = "SELECT ROUND(bucket_quantile * 100, 1) AS p, ROUND(bucket_timer_high / 1000000, 3) AS us FROM performance_schema.events_statements_histogram_global"
+	TRUNCATE_QUERY = "TRUNCATE TABLE performance_schema.events_statements_histogram_global"
+)
+
+type Global struct {
+	db *sql.DB
+	// --
+	optional      map[string]bool      // keyed on level
+	percentiles   map[string][]float64 // keyed on level
+	truncateTable map[string]bool      // keyed on level
+	available     bool
+}
+
+var _ blip.Collector = &Global{}
+
+func NewGlobal(db *sql.DB) *Global {
+	return &Global{
+		db:            db,
+		optional:      map[string]bool{},
+		percentiles:   map[string][]float64{},
+		truncateTable: map[string]bool{},
+		available:     true,
+	}
+}
+
+// Domain returns the Blip metric domain name (DOMAIN const).
+func (c *Global) Domain() string {
+	return DOMAIN
+}
+
+// Help returns the output for blip --print-domains.
+func (c *Global) Help() blip.CollectorHelp {
+	return blip.CollectorHelp{
+		Domain:      DOMAIN,
+		Description: "Collect metrics for all queries",
+		Options: map[string]blip.CollectorHelpOption{
+			OPT_RESPONSE_TIME_OPTIONAL: {
+				Name:    OPT_RESPONSE_TIME_OPTIONAL,
+				Desc:    "If collecting response time metrics is optional. This will fail if it is required but not available",
```

Comment:
I need to think more about this kind of option (here and in other collectors). We want something like a 3-part option:
* Required (error in Prepare if not)
* Optional (try but no error on failure)
Granted, we kind of have that now... I'm just thinking "optional" is not the best option name for this 🤔

Reply:
`Optional` is kind of confusing at the first glance. Maybe this option can be something more explicit like `ignore-unavailable-error`.